# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/media/webrtc/jsapi/RTCRtpSender.h
# Commit: fef019ffca5f
# Full Hash: fef019ffca5f8a98a0eab2f63ff7aa148afb2706
# Author: Byron Campen [:bwc] <docfaraday@gmail.com>
# Date: 2022-12-22 21:34:53
# Regressor Bug: 1800136
# File Overlap Count: 1
# Description:
#   Bug 1800136: Make RTCRtpSender::GetStats get ICE stats. r=ng
#   
#   Also fix bugs where we would get duplicate ICE stats. This mostly mattered to
#   WebrtcGlobalInformation since RTCStatsReportInternal does not automatically
#   remove duplicate ids like RTCStatsReport (RTCStatsReport is maplike).
# ==============================================================================

diff -r ab6eb1cfa9a0 -r fef019ffca5f dom/media/webrtc/jsapi/RTCRtpSender.h
--- a/dom/media/webrtc/jsapi/RTCRtpSender.h	Thu Dec 22 15:29:02 2022 +0000
+++ b/dom/media/webrtc/jsapi/RTCRtpSender.h	Thu Dec 22 15:29:02 2022 +0000
@@ -71,7 +71,8 @@
       ErrorResult& aRv);
 
   nsPIDOMWindowInner* GetParentObject() const;
-  nsTArray<RefPtr<RTCStatsPromise>> GetStatsInternal();
+  nsTArray<RefPtr<RTCStatsPromise>> GetStatsInternal(
+      bool aSkipIceStats = false);
 
   // This would just be stream ids, except PeerConnection.jsm uses GetStreams
   // to implement the non-standard RTCPeerConnection.getLocalStreams. We might
@@ -162,6 +163,7 @@
   bool mSimulcastEnvelopeSet = false;
   Maybe<RTCRtpSendParameters> mLastReturnedParameters;
   RefPtr<MediaPipelineTransmit> mPipeline;
+  RefPtr<MediaTransportHandler> mTransportHandler;
   RefPtr<RTCRtpTransceiver> mTransceiver;
   nsTArray<RefPtr<DOMMediaStream>> mStreams;
   bool mHaveSetupTransport = false;
